blog_posts/2024-09-12-optimisations-text-more.markdown

```diff
@@ -0,0 +1,142 @@
+    <source src="/2024-09-12-optimisations-text-more/edit-avm1-value.mp4" type="video/mp4">
+</video>
+
+### Android Application
+For a while now, @torokati44 has been maintaining an unofficial Android application for Ruffle, which was pretty barebones but was already quite popular despite not being released anywhere. We've decided to bring it under the Ruffle umbrella and have officially adopted it as our Android application.
+@dinnerbone has overhauled the codebase and UI for it, giving it support for most Flash content - even online games like Transformice work!
+However, it's still too early for us to feel comfortable publishing it to the Play Store, as there's a lot of UX that needs improving. Help is very much appreciated from any Android developers!
+
+We've seen that there's already a bunch of apps out there which package Ruffle, and that's pretty cool! However, they seem to all use the web version of Ruffle, which leaves a lot of performance on the table. We're hoping that by using the native codebase, we'll end up with really decent performance and stability... finally putting Flash back on mobile phones after all this time!
+*(perhaps iOS too, now that they've relaxed their rules about emulators... but we'd need an iOS developer to champion that!)*
+
+### Browser Extension
+There's been a lot of improvements to the extension codebase, and some bug fixes here and there. The major points here though are:
+
+- @danielhjacobs has upgraded us fully to Manifest V3 on Firefox, with a new Onboarding screen for permission requests.
+- @danielhjacobs has also brought back "SWF takeover" (loading a SWF url directly, and have it play in Ruffle instead of downloading) using the upcoming header detection feature of `declarativeNetRequest` in Manifest V3. This API only works in latest Chromium currently.
+- @riku-42 has added support for 4399.com (a Chinese flash game site) with our extension by adding better Flash detection workarounds.
+
+### Web Improvements
+There has also been a lot of improvements under the hood here for code quality, and especially preparation for releasing a stable version. There's a few notable highlights though:
+
+- Thanks to contributions from @WumboSpasm, the general look and feel of the UI on web is much improved. This includes support for dark mode, which will match the browsers settings.
+- @dinnerbone has been working on stabilizing our JavaScript API, and also making a `ruffle-core` package that anyone can use as an npm dependency, instead of relying on our prebuilt zip packages. This work is ongoing but should be completed soon!
+- @dinnerbone has improved `ExternalInterface` support, allowing support for games like Club Penguin which interacted with the page heavily.
+
+### Rendering Improvements
+@Aaron1011 has fixed a bunch of bugs relating to Stage3D, and improved PixelBender support.
+
+@moulins has found and fixed lots of smaller issues with our Filter implementations, and provided better tests for them.

```

Comment:
I don't think filter here should be capitalized? "filter effect" might be more clear.